src/core/gltf2importer/meshparser_utils.cpp

```diff
@@ -382,27 +382,25 @@ Qt3DRender::QAttribute *createTangentAttribute(Qt3DRender::QGeometry *geometry,
 bool geometryIsGLTF2Valid(Qt3DRender::QGeometry *geometry)
 {
     const auto &attributes = geometry->attributes();
-    const bool vertexBaseTypesAreValid = std::find_if(std::begin(attributes), std::end(attributes),
-                                                      [](Qt3DRender::QAttribute *attribute) {
-                                                          const auto validVertexBaseTypes = validVertexBaseTypesForAttribute(attribute->name());
-                                                          const auto vertexBaseType = attribute->vertexBaseType();
-                                                          return std::find_if(std::begin(validVertexBaseTypes),
-                                                                              std::end(validVertexBaseTypes),
-                                                                              [vertexBaseType](Qt3DRender::QAttribute::VertexBaseType validVertexBaseType) {
-                                                                                  return vertexBaseType == validVertexBaseType;
-                                                                              }) == std::end(validVertexBaseTypes);
-                                                      }) == std::end(attributes);
-
-    const bool vertexSizesAreValid = std::find_if(std::begin(attributes), std::end(attributes),
-                                                  [](Qt3DRender::QAttribute *attribute) {
-                                                      const auto validVertexSizes = validVertexSizesForAttribute(attribute->name());
-                                                      const auto vertexSize = attribute->vertexSize();
-                                                      return std::find_if(std::begin(validVertexSizes),
-                                                                          std::end(validVertexSizes),
-                                                                          [vertexSize](uint validVertexSize) {
-                                                                              return vertexSize == validVertexSize;
-                                                                          }) == std::end(validVertexSizes);
-                                                  }) == std::end(attributes);
+    const bool vertexBaseTypesAreValid = [&] () {
```

Comment:
I dont find this easier to read. Added a comment in the conversation.

Reply:
What do you want me to do exactly ? The commit is about fixing the unit tests which are currently broken since the introduction of the geometryIsGLTF2Valid leaving the CI in a broken state currently. For the tests to pass, as indicated in the commit message, the first two entries are necessary. I do agree that the last entry in the commit message is more voicing my personal opinion than factuals, I've updated the commit message accordingly and reworked the code following your comments. Please let's try not to hold this patch in review too long